lib/myFTClient.js

```diff
@@ -25,266 +25,437 @@ const digestProperties = Object.assign({"type": "daily", "timezone": "Europe/Lon
+ * @param {Object} preference - an object representing the user's preference.
  *        at a minimum this should be type and timezone. For example:
- *        { type: "daily", timezone:"Europe/London", byTool: "KAT",
+ *        {type: "daily", timezone:"Europe/London", byTool: "KAT",
  *          byUser: "8619e7a0-65b7-446b-9931-4197b3fe0cbf"}
- * @param {object} [options] - additional options {supressEvents: true|false, waitForPurge: true|false }
+ * @param {Object} [options] - additional options {supressEvents: true|false, waitForPurge: true|false }
  *        default behaviour not to supress event generation and wait for a cache purge
- * @return {Promise} EmailDigestPreference json structure
- * @throws {error} statusError - if something goes wrong
+ * @return {Promise} response -  EmailDigestPreference json structure
+ * @throws {Error} statusError - if something goes wrong
  */
 function setEmailDigestPreference(uuid, preference, options) {
-	const operation = 'setEmailDigestPreferences';
-	return addRelationships('user', uuid, 'preferred', 'preference', 'email-digest', preference, options)
-		.then((res)=>{
-			return helpers.parseJsonRes(res, `${operation} for user ${uuid}`)
-				.then ((edp)=>{
-					log.debug({operation, uuid, edp:JSON.stringify(preference)});
-					return edp;
-				});
-		});
+	const operation = 'myFTClient.setEmailDigestPreferences';
+  const prefStr = JSON.stringify(preference);
+  const optStr = JSON.stringify(options);
+  log.debug({operation, uuid, preference: prefStr, options: optStr});
+	return _addRemoveRelationships('POST', 'user', uuid, 'preferred', 'preference', 'email-digest', preference, options)
+    .then(res => helpers.parseJsonRes(res, `${operation} for user ${uuid}`))
+		.then(res => {
+      log.debug({operation, uuid, preference: prefStr, options: optStr, res: JSON.stringify(res)});
+      return res;
+    });
 }
 
 /**
- * Gets the users with EmailDigestPreferences for given licence
- * @param {string} uuid - of the licence
- * @return {Promise} array of user json structures
- */
-//function getUsersWithEmailDigestPreference(uuid) {// TODO: WIP
-//	const operation = 'usersWithEmailDigestPreferences';
-//	return getScopedRelationships('license', uuid, 'preference', 'email-digest', 'preferred', 'user')
-//		.then((res)=>{
-//			return helpers.parseJsonRes(res, `${operation} for licence ${uuid}`)
-//				.then ((users)=>{
-//					log.debug({operation, uuid, count: users.length, users});
-//					return users;
-//				});
-//		});
-//}
-
-/**
  * Gets the Concepts followed by a user
- * @param {string} uuid - of the user
- * @return {Promise} array of Concepts followed
+ * @param {String} uuid - of the user
+ * @return {Promise} response - array of Concepts followed
  */
-function getConceptsFollowedByUser(uuid){
-	return getAllNodeItems(uuid, "user", 'followed', 'concept');
+function getConceptsFollowedByUser(uuid) {
+  return _getAllNodeItems('user', uuid, 'followed', 'concept');
```

Comment:
There's a number of areas where the node name (i.e. user, group, etc.) is hard coded as a string when calling different methods such as this one. For the sake of maintainability could we move these into a set of constants for this module and pass those to the methods instead?